phoenix-core-client/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -4498,6 +4509,13 @@ public void upgradeSystemTables(final String url, final Properties props) throws
                         "Snapshots for system tables created so far: {}",
                         systemTableToSnapshotMap);
                 }
+                if (acquiredBlockMutexLock) {
+                    try {
+                        releaseUpgradeBlockMutex();
```

Comment:
I'm not sure we want to release this mutex automatically.

This block was created manually, and the upgrade may have failed, which we have no easy way of checking from code.